client/ionic/src/pages/About.tsx

```diff
@@ -11,111 +12,83 @@ import {
-                  with one another (within about 6 feet or two meters) through
-                  respiratory droplets produced when an infected person coughs
-                  or sneezes.
-                </IonCardSubtitle>
-              </IonCardHeader>
-              <IonCardContent>
-                <IonIcon size="large" icon={personAddOutline}></IonIcon>
-              </IonCardContent>
-              <IonCardContent>
-                <IonButton href="/menu" shape="round">
-                  Learn More
-                </IonButton>
-              </IonCardContent>
-            </IonCard>
-          </IonSlide>
-          <IonSlide>
-            <IonCard>
-              <IonCardHeader>
-                <IonCardTitle>Clean Your Hands</IonCardTitle>
-                <IonCardSubtitle>
-                  <b>Wash your hands often</b> with soad and water for at least
-                  20 sconds, especially after you have been in a public place,
-                  or after blowing your nose, coughing or sneezing.
-                </IonCardSubtitle>
-                <IonCardSubtitle>
-                  <b>Avoid touching your eyes, nose, and mouth</b> with unwashed
-                  hands.
-                </IonCardSubtitle>
-              </IonCardHeader>
-              <IonCardContent>
-                <IonIcon size="large" icon={handLeftOutline}></IonIcon>
-                <IonIcon size="large" icon={handRightOutline}></IonIcon>
-              </IonCardContent>
-              <IonCardContent>
-                <IonButton href="/menu" shape="round">
-                  Learn More
-                </IonButton>
-              </IonCardContent>
-            </IonCard>
-          </IonSlide>
-          <IonSlide>
-            <IonCard>
-              <IonCardHeader>
-                <IonCardTitle>Avoid Close Contact</IonCardTitle>
-                <IonCardSubtitle>
-                  <b>Avoid close contact</b> with people who are sick.
-                </IonCardSubtitle>
-                <IonCardSubtitle>
-                  <b>Maintain distance between yourself and other people</b> if
-                  COVID-19 is spreading in your community. This is especially
-                  important for people who are at a higher risk of getting very
-                  sick, including the elderly.
-                </IonCardSubtitle>
-              </IonCardHeader>
-              <IonCardContent>
-                <IonIcon size="large" icon={peopleOutline}></IonIcon>
-              </IonCardContent>
-              <IonCardContent>
-                <IonButton href="/menu" shape="round">
-                  Learn More
-                </IonButton>
-              </IonCardContent>
-            </IonCard>
-          </IonSlide>
-        </IonSlides>
+        {flow.content && flow.content.screens && (
+          <IonSlides pager={true}>
+            {flow.content.screens.map(screen => {
+              switch (screen.type) {
```

Comment:
Rather than a `switch` statement here, it may be cleaner to define a dictionary of components

```js
let screens = {
   "TextImage": TextImageComponent
};
```

and then access the correct component using `screens[screen.type]`?

Reply:
agree with refactoring as noted in the TODO in the code :)

I'd suggest a thin `Screen` component that gets passed the `screen` object as a prop and handles the logic of choosing a subcomponent (or falling back on unsupported type) like `TextImageComponent` which would then handle the rendering of all of its children

Reply:
Yes, added that to the TODO to Refactor this out to separate Flow components.

Each of those named screens will correspond to a screen archetype to be defined by PM/Design, but they haven't gotten around to it.